scripts/train_lm.py

```diff
@@ -196,19 +196,17 @@ def FormatMetaparameters(metaparameters, num_train_sets):
 
     return ','.join(out_param)
 
-# this function checks whether there are repeating values in parameters(excluding 
-# the first #number numbers, which are not data-type weights) and if found, return
-# the index of the first pair repeating values 
+# this function checks repeating values in parameters and returns the index
+# of the first pair of repeating values if any is found 
 def FindRepeatingValue(parameters):
-#assert (number >= 1) and (number < len(parameters))
     for i in range(0, len(parameters)):
         for j in range(i + 1, len(parameters)):
             if parameters[i] == parameters[j]:
                 return(i, j)
     return (-1, -1)
 
-# this function rerounds repeating values until they are different and return the values after rerounding
-# if they are indeed the same (collison), exit with a warning (this case can be very rare) 
+# this function rerounds repeating values until they are different and returns the values after rerounding.
+# if they are indeed the same (collison), it exits with a warning (this case can be very rare) 
 def AdjustValues(number1, number2, index1, index2, ref_number1, ref_number2):
```

Comment:
I see. So we only let the script die when there are 1) identical
data-directory weights or 2) identical data-type weights for the same order
or 3) exact 0 or 1? I think any of the three cases indicates something must
be wrong and thus we'd better let the script exit with error when any of
them happens. What do you think?
Ke


On Sun, Aug 28, 2016 at 6:54 PM, Daniel Povey <notifications@github.com>
wrote:

> In scripts/train_lm.py
> <https://github.com/danpovey/pocolm/pull/52#discussion_r76540845>:
>
> >      for i in range(0, len(parameters)):
> >          for j in range(i + 1, len(parameters)):
> >              if parameters[i] == parameters[j]:
> >                  return(i, j)
> >      return (-1, -1)
> >
> > -# this function rerounds repeating values until they are different and return the values after rerounding
> > -# if they are indeed the same (collison), exit with a warning (this case can be very rare)
> > +# this function rerounds repeating values until they are different and returns the values after rerounding.
> > +# if they are indeed the same (collison), it exits with a warning (this case can be very rare)
> >  def AdjustValues(number1, number2, index1, index2, ref_number1, ref_number2):
>
> I think if two values are the same (collision), it would be better to just
> print them with full precision rather than die.
> The reason is that collisions are only disallowed in certain
> circumstances: for the same order of discounting parameter, we cannot have
> d1 == d2 or d2 == d3. But otherwise, collisions are quite possible. Making
> this cause the script to die is a bad idea.
>
> And we cannot have two data-directory weights be identical [IIRC]. But
> otherwise, collisions
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/danpovey/pocolm/pull/52/files/732e304fd3750c911ce87076cd76aa546090ed2a..e95b774e86bd4ee833b3430ec32a3a052059e26a#r76540845>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ANVxSrRvwF7_vkt7jgPcLer-4Hons9Qyks5qkhGJgaJpZM4Jl6Bt>
> .
>



-- 
Ke Li
Dept. of Electrical and Computer Engineering
Johns Hopkins University
Email: kli26@jhu.edu